src/getting-started.md

```diff
@@ -0,0 +1,470 @@
+```shell
+cp config.toml.example config.toml
+```
+
+Then, edit `config.toml`. You may want to search for, uncomment, and update
+the following settings:
+
+- `debug = true`: enables debug symbols and `debug!` logging, takes a bit longer to compile.
+- `incremental = true`: enables incremental compilation of the compiler itself,
+  which can significantly speed things up.  This is turned off by default
+  because it's technically unsound; sometimes it will cause weird crashes.
+  Also, it can consume a lot of disk space. This has the same effect as the
+  `-i` or `--incremental` flags.
+- `llvm-config`: enables building with system LLVM. [See this chapter][sysllvm]
+  for more info. This avoids building LLVM, which can take a while.
+
+[sysllvm]: ./building/suggested.html#building-with-system-llvm
+
+### `./x.py` Intro
+
+`rustc` is a _bootstrapping_ compiler, which means that it is written in Rust. So where do you
+get the original compiler from? We use the current beta compiler
+to build the compiler. Then, we use that compiler to build itself. Thus,
+`rustc` has a 2-stage build. You can read more about bootstrapping
+[here][boot], but you don't need more to contribute.
+
+[boot]: ./building/bootstrapping.md
+
+We have a special tool `./x.py` that drives this process. It is used for
+compiling the compiler, the standard libraries, and `rustdoc`. It is also used
+for driving CI and building the final release artifacts.
+
+Unfortunately, a proper 2-stage build takes a long time depending on your
+hardware, but it is the only correct way to build everything (e.g. it's what
+the CI and release processes use). **However, in most cases, you can get by
+without a full 2-stage build**. In the following section, we give instructions
+for how to do "the correct thing", but then we also give various tips to speed
+things up.
+
+### Building and Testing `rustc`
+
+To do a full 2-stage build of the whole compiler, you should run this (after
+updating `config.toml` as mentioned above):
+
+```sh
+./x.py build
+```
+
+In the process, this will also necessarily build the standard libraries, and it
+will build `rustdoc` (which doesn't take too long).
+
+To build and test everything:
+
+```sh
+./x.py test
+```
+
+For most contributions, you only need to build stage 1, which saves a lot of time:
+
+```shell
+# Build the compiler (stage 1)
+./x.py build --stage 1
+
+# Subsequent builds
+./x.py build --stage 1 --keep-stage 1
+```
+
+This will take a while, especially the first time. Be wary of accidentally
+touching or formatting the compiler, as `./x.py` will try to recompile it.
+
+**NOTE**: The `--keep-stage 1` will _assume_ that the stage 0 standard library
+does not need to be rebuilt, which is usually true, which will save some time.
+However, if you are changing certain parts of the compiler, this may lead to
+weird errors. Feel free to ask on [zulip][z] if you are running into issues.
+
+This runs a ton of tests and takes a long time to complete. If you are
```

Comment:
I am unsure that the tests take all available CPU cores to run, and in my experience I was under the impression that the tests ran faster if I passed the `--jobs` flag. I am sure of nothing though, I should probably dive into the sources for the bootstrap to find out.

Reply:
TMK, compiletest will use all available CPUs and run as many parallel tasks as possible. I might be wrong though. I just remember that on my laptop compiletest will max out CPU.

Reply:
>  the tests ran faster if I passed the --jobs flag

This depends on what you have `codegen-units` set to in `config.toml`. I have it set to `codegen-units = 0` but I think the default might be 1.